interfaces/apparmor/apparmor.go

```diff
@@ -115,7 +116,24 @@ func loadProfiles(fnames []string, cacheDir string, flags aaParserFlags) error {
 	}
 	args = append(args, fnames...)
 
-	output, err := exec.Command("apparmor_parser", args...).CombinedOutput()
+	parser, internal, err := apparmor_sandbox.FindAppArmorParser()
+	if err == nil || !internal {
+		// if we couldn't find the parser with apparmor_sandbox
+		// then fall-back to trying to find one in the current PATH
+		// - same for if we are not using the internal
+		// apparmor_parser so that we can support a mocked parser
+		// during tests
+		parser = "apparmor_parser"
+	} else {
+		// when using the internal apparmor_parser also use it's
+		// own configuration and includes etc
+		prefix := strings.TrimSuffix(parser, "apparmor_parser")
+		args = append(args, "--config-file")
+		args = append(args, filepath.Join(prefix, "/apparmor/parser.conf"))
+		args = append(args, "-b")
+		args = append(args, filepath.Join(prefix, "/apparmor.d"))
```

Comment:
does this mean that the apparmor config that is used from the snapd snap is always read-only? i..e. one can't modify the parser config or add/modify an abstraction without rebuilding the snapd snap? That seems a bit unfortunate and may have implications for folks who are currently doing this to support NFS homes or other random things we don't properly support but can be supported by changing the apparmor abstractions on their system

Reply:
Yes this is perhaps unfortunate in that case, but I wanted to ensure that if we are invoking the snapd internal apparmor_parser that we then also use the associated snapd-internal apparmor config/abstractions etc. I guess the other way to do this would be to have the snapd snap copy out these files to somewhere writable when it is installed (/var/lib/snapd/apparmor/ ?) and then use this path instead here so that users can change these files if desired.

Reply:
Probably for now it's fine to have static config etc. from $SNAP, but this is something to think about and call out, if not eventually change. I think the proposal to have some hook/process inside snapd copy the static read-only files out of the snapd snap and into somewhere like /var/lib/snapd/apparmor is very reasonable FWIW

Reply:
As far as abstractions are concerned, multiple include paths can be specified, so that part should be easier to solve.

Reply:
The apparmor_parser docs seem to indicate that `-b` and `-I` affect different types of includes (relative paths vs. absolute). It also sounds like there can only be one "base" directory.

Reply:
if I understand this is a functionality regression so we do need an answer before this goes to stable? or am I confused?

Reply:
Yes this is a functionality regression (ie that now snapd will always use a static, read-only apparmor configuration which it ships itself whereas before it would use the one on the host which could then be modified etc as needed by a local admin).

But I am not sure where if anywhere we actually document this functionality - if we do publicly document this somewhere then we would want to make sure this PR can support that still in some form. Also there are advantages to shipping this static config as it means snapd should be more robust since an admin can't inadvertently mess up the global apparmor config and hence break snapd confinement etc - so I suppose we need to make a choice here whether this is something we still want to support and if so, how best to do it.

Reply:
Well I don't think we have any public docs on doing this, but there are various forum posts where folks have figured out that editing these config files enables them to use snapd with their specific setup, see for example https://forum.snapcraft.io/t/how-can-i-use-snap-when-i-dont-use-home-user/3352 but I don't think those comments qualify as "supported documented features".

Reply:
As @anonymouse64 said, we've pointed people to the AppArmor tunables as ways to get snaps to work on their systems in the past. If we're changing how policies are compiled, it would be nice if those customisations continued to work. If that's not possible, I think we'd need to be able to provide a new advice to get snapd working in those environments.

Would it make sense to symlink `.../etc/apparmor.d/tunables` (or some files or directories under there) to the equivalent locations in the real `/etc`? This would require that there still be an AppArmor installation on the host system, but that's probably going to be the base for any cases where we want to install policies.

I guess the only question is whether the default tunables from old versions of AppArmor are going to be compatible with the rest of the `apparmor.d/abstractions` rules from the new AppArmor.

Reply:
I don't think we can reliably mix-and-match configurations between what we ship and what may be on the host. I think the better option is to copy out our known-good static configuration to a writable location (`/var/lib/snapd/apparmor/` ?) and then document this as where users need to customise rather than trying to include any existing host apparmor config bits. Whilst this then means they have to configure the same thing in two places, this should ensure there is less chance that some incompatibility between the snapd and the hosts apparmor config/abstractions etc causes things to break.

Reply:
I think it would be okay if we agreed on a timeframe when this feature could be worked on as well as who would work on implementing the feature to copy this config directory out and point apparmor_parser at that writable directory and merge this PR as-is with the understanding that the feature would not get released to stable until that feature was implemented and properly tested with i.e. spread tests and then also officially documented as something we commit to continuing to maintain.

Reply:
I think it's okay if in the short term we break the small quantity of users on the edge channel who may rely on the existing functionality as long as we have a documented, supported way forward for all the users currently using this workaround on the stable channel, though we no idea how many users that is

Reply:
See the addition of https://github.com/snapcore/snapd/pull/9939/commits/9b2a2866cc6024a11887011a21e435b95be84ce1 which allows the site-local tunables to be customised via a `/var/lib/snapd/apparmor/snap-tunables` file which will get included in all generated profiles IIF it exists (ie this has to be manually created by a site admin, we don't ship one by default). This should allow to alleviate any concerns with the read-only apparmor config/tunables etc inside the snapd snap as a result of this change.